fuel-forge/src/main/java/com/github/kittinunf/fuel/forge/FuelForge.kt

```diff
@@ -28,7 +28,13 @@ inline fun <reified T : Any> Request.responseObject(noinline deserializer: JSON.
 inline fun <reified T : Any> Request.responseObjects(noinline deserializer: JSON.() -> DeserializedResult<T>) = response(forgesDeserializerOf(deserializer))
 
 fun <T : Any> forgeDeserializerOf(deserializer: JSON.() -> DeserializedResult<T>) = object : ResponseDeserializable<T> {
-    override fun deserialize(content: String): T? = Forge.modelFromJson(content, deserializer).component1()
+    override fun deserialize(content: String): T? {
+        val forge =  Forge.modelFromJson(content, deserializer)
+        forge.component2()?.let {
```

Comment:
Could we do it? Like, I think the better way to do this is 
```suggestion
when (deserializedResult) {
  is Success -> ...
  is Failure -> { throw error }
}
```